resource-aggregate/cqrs/projection/projection_test.go

```diff
@@ -156,6 +159,50 @@ func prepareResourceStateEventstore(t *testing.T) *mockEvents.MockEventStore {
 	return eventstore
 }
 
+func TestResourceProjectionTestLoadParallelModels(t *testing.T) {
+	eventstore := mockEvents.NewMockEventStore()
+	resourceChangedEventMetadata := makeEventMeta("", 0, 0)
+	numDevices := 350
+	numResources := 500
+	numParallelRequests := 6
+
+	for i := 0; i < numDevices; i++ {
+		for j := 0; j < numResources; j++ {
+			resourceID := commands.NewResourceID(fmt.Sprintf("dev-%v", i), fmt.Sprintf("res-%v", j))
+			eventstore.Append(resourceID.GetDeviceId(), resourceID.ToUUID(), mockEvents.MakeResourceStateSnapshotTaken(resourceID, &events.ResourceChanged{Content: &commands.Content{}, EventMetadata: resourceChangedEventMetadata}, makeEventMeta("a", 0, 0), mockEvents.MakeAuditContext("userId", "2")))
+		}
+	}
+
+	ctx := context.Background()
+	p, err := NewProjection(
+		ctx,
+		"test",
+		eventstore,
+		nil,
+		func(ctx context.Context, groupID, aggregateID string) (cqrsEventStore.Model, error) {
+			return events.NewResourceLinksSnapshotTaken(), nil
+		},
+	)
+	require.NoError(t, err)
+
+	err = p.cqrsProjection.Project(ctx, nil)
+	require.NoError(t, err)
+
+	var wg sync.WaitGroup
+	wg.Add(numParallelRequests)
+	t.Logf("starting %v requests\n", numParallelRequests)
+	for i := 0; i < numParallelRequests; i++ {
+		go func(v int) {
+			defer wg.Done()
+			n := time.Now()
+			p.cqrsProjection.Models(nil, func(m cqrsEventStore.Model) (wantNext bool) { return true })
+			diff := -1 * time.Until(n)
+			t.Logf("%v models time %v\n", v, diff)
```

Comment:
Is this intended to trigger race detector in case of a problem? Because there is no verification here, only a log.

Reply:
this test is only for manual measure time